addon/components/high-charts.js

```diff
@@ -73,7 +73,11 @@ export default Component.extend({
         return chartSeriesToRemove.push(series);
       }
 
-      series.setData(contentSeries.data, false);
+      if (Object.keys(contentSeries).length > 1) {
```

Comment:
This seems like it is going to evaluate to true the majority of use cases. Most `series` have at least two keys:
```js
{
  name: 'Jane',
  data: [5, 0, 20]
}
```

As you mentioned `series.update` is an expensive operation so I'd like to avoid it whenever possible. 

In the dummy app in this repo we have an interactive line chart example for manually testing the "update data" feature. I tested that with your change and `series.update` always gets called. 

Perhaps we can do a diff between `contentSeries` and `chart.series`. If only the `data` value changed then call `setData`. Else call `series.update`.